### PR TITLE
[MAINTENANCE] Check filepath existence when evaluating public API report

### PR DESCRIPTION
**Pull request by cdkini**
We should use a rich type vs a string and always check for filepath existence. 

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `ruff format` + `ruff check`)
- [x] Appropriate tests and docs have been updated

For more information about contributing, visit our [community resources](https://docs.greatexpectations.io/docs/core/introduction/community_resources#contribute-code-or-documentation).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: docs/sphinx_api_docs_source/public_api_missing_threshold.py [hunk 1]
```diff
@@ -4,165 +4,781 @@
...
-    "File: great_expectations/expectations/core/expect_multicolumn_sum_to_equal.py Name: ExpectMulticolumnSumToEqual",
-    "File: great_expectations/expectations/core/expect_column_proportion_of_unique_values_to_be_between.py Name: ExpectColumnProportionOfUniqueValuesToBeBetween",
-    "File: great_expectations/expectations/core/expect_column_unique_value_count_to_be_between.py Name: ExpectColumnUniqueValueCountToBeBetween",
-    "File: great_expectations/expectations/core/expect_column_values_to_be_unique.py Name: ExpectColumnValuesToBeUnique",
-    "File: great_expectations/expectations/core/expect_compound_columns_to_be_unique.py Name: ExpectCompoundColumnsToBeUnique",
-    "File: great_expectations/expectations/core/expect_select_column_values_to_be_unique_within_record.py Name: ExpectSelectColumnValuesToBeUniqueWithinRecord",
-    "File: great_expectations/expectations/core/expect_column_min_to_be_between.py Name: ExpectColumnMinToBeBetween",
+    PrintableDefinition(
+        file=pathlib.Path("great_expectations/_docs_decorators.py"), name="add"
+    ),
```
Comment (cdkini) [inline docs/sphinx_api_docs_source/public_api_missing_threshold.py:14]: I'm of the opinion that a rich type is more valuable here - we also want to ensure that filepaths stay accurate and this type does that.

Comment (NathanFarmer) [reply docs/sphinx_api_docs_source/public_api_missing_threshold.py:14]: Hard agree that parsing a string was not preferable

**Review by NathanFarmer — COMMENTED** [review]
Isn't there a test threshold that can now be lowered?

**Comment by cdkini** [comment]
@NathanFarmer I'm honestly not too sure - do you know where I can find that? I thought the threshold was dynamic based on the length of the lists?

**Comment by NathanFarmer** [comment]
> @NathanFarmer I'm honestly not too sure - do you know where I can find that? I thought the threshold was dynamic based on the length of the lists?

@cdkini yeah that does appear to be the case actually. Please re-request me once tests get passing on this.

**Review by NathanFarmer — APPROVED** [review]

**Pull request merged**